Jenkinsfile

```diff
@@ -97,45 +93,72 @@ def buildGrakn() {
 //Only run validation master/stable
 if (env.BRANCH_NAME in ['master', 'stable']) {
     properties([buildDiscarder(logRotator(numToKeepStr: '30', artifactNumToKeepStr: '7'))])
+
     node {
+        slackGithub "Build started"
+
         String workspace = pwd()
         checkout scm
 
-        slackGithub "Build started"
-
-        timeout(60) {
-            stage('Run the benchmarks') {
-                sh "mvn clean test --batch-mode -P janus -Dtest=*Benchmark -DfailIfNoTests=false -Dmaven.repo.local=${workspace}/maven -Dcheckstyle.skip=true -Dfindbugs.skip=true -Dpmd.skip=true"
-                archiveArtifacts artifacts: 'grakn-test/test-integration/benchmarks/*.json'
+        stage('Build Grakn') {
+            withScripts(workspace) {
+                buildGrakn()
             }
-        }
 
-        for (String moduleName : integrationTests) {
-            runIntegrationTest(workspace, moduleName)
-        }
+            archiveArtifacts artifacts: "grakn-dist/target/grakn-dist*.tar.gz"
 
-        slackGithub "Periodic Build Success", "good"
+            // Stash the built distribution so other nodes can access it
+            stash includes: 'grakn-dist/target/grakn-dist*.tar.gz', name: 'dist'
+        }
     }
-}
 
-// Deploy long-running instance on stable branch
-if (env.BRANCH_NAME == 'stable') {
-    node {
-        String workspace = pwd()
-        checkout scm
+    jobs = [
+        benchmarks: {
+            node {
+                String workspace = pwd()
+                checkout scm
+                unstash 'dist'
+
+                timeout(60) {
+                    stage('Run the benchmarks') {
+                        sh "mvn clean test --batch-mode -P janus -Dtest=*Benchmark -DfailIfNoTests=false -Dmaven.repo.local=${workspace}/maven -Dcheckstyle.skip=true -Dfindbugs.skip=true -Dpmd.skip=true"
+                        archiveArtifacts artifacts: 'grakn-test/test-integration/benchmarks/*.json'
+                    }
+                }
+            }
+        }
+    ];
 
-        stage('Build Grakn') {
-            withScripts(workspace) {
-                buildGrakn()
+    for (String moduleName : integrationTests) {
+        jobs[moduleName] = {
+            node {
+                String workspace = pwd()
+                checkout scm
+                unstash 'dist'
+
+                runIntegrationTest(workspace, moduleName)
             }
         }
+    }
 
-        stage('Deploy Grakn') {
-            sshagent(credentials: ['jenkins-aws-ssh']) {
-                sh "scp -o StrictHostKeyChecking=no grakn-dist/target/grakn-dist*.tar.gz ubuntu@${LONG_RUNNING_INSTANCE_ADDRESS}:~/"
-                sh "scp -o StrictHostKeyChecking=no scripts/repeat-query ubuntu@${LONG_RUNNING_INSTANCE_ADDRESS}:~/"
-                ssh "'bash -s' < scripts/start-long-running-instance.sh"
+    parallel(jobs);
+
+    node {
```

Comment:
You can remove it as there's a nested 'node' clause here.

Reply:
This node is actually not inside another. I think the `parallel` part has to be executed outside of a node, so I leave the first node to run then - then I have to make a new one to deploy at the end.

Reply:
There's another node clause on 153 though.

Reply:
Ohhh! I see it, thanks.